File: front/src/types/camp.ts
export interface CampProject {
  id: number;
  term: string;
  image_url?: string | null;
  team_name?: string | null;
  description?: string | null;
  content?: string | null;
  subject?: string | null;
  leader_name?: string | null;
  member_names?: string | null;
  display_order?: number | null;
  is_visible: boolean;
  created_at: string;
  updated_at: string;
}

export interface MainProjectRow {
  id: number;
  teamName: string;
  projectName: string;
  subject: string;
  notionUrl: string | null;
}

export interface CampProjectCard {
  id: number;
  term: string;
  title: string;
  subtitle: string;
  imgUrl: string;
  topic: string;
  leader: string;
  members: string;
  description: string;
}

export const mapCampProjectToRow = (project: CampProject): MainProjectRow => ({
  id: project.id,
  teamName: project.team_name || '-',
  projectName: project.description || '-',
  subject: project.subject || '-',
  notionUrl: project.content?.startsWith('http') ? project.content : null,
});

export const mapCampProjectToCard = (project: CampProject): CampProjectCard => ({
  id: project.id,
  term: project.term,
  title: project.team_name || '팀',
  subtitle: project.description || '',
  imgUrl: project.image_url || '#EED270',
  topic: project.subject || '',
  leader: project.leader_name || '',
  members: project.member_names || '',
  description: project.content || '',
});

File: app/schemas/camp.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CampProjectBase(BaseModel):
    term: str
    image_url: Optional[str] = None
    team_name: Optional[str] = None
    description: Optional[str] = None
    content: Optional[str] = None
    subject: Optional[str] = None
    leader_name: Optional[str] = None
    member_names: Optional[str] = None
    display_order: Optional[int] = None
    is_visible: bool = True


class CampProjectCreate(CampProjectBase):
    pass


class CampProjectUpdate(BaseModel):
    term: Optional[str] = None
    image_url: Optional[str] = None
    team_name: Optional[str] = None
    description: Optional[str] = None
    content: Optional[str] = None
    subject: Optional[str] = None
    leader_name: Optional[str] = None
    member_names: Optional[str] = None
    display_order: Optional[int] = None
    is_visible: Optional[bool] = None


class CampProjectResponse(CampProjectBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True